Lib/test/test_urllib2net.py

```diff
@@ -143,6 +146,36 @@ def test_ftp(self):
             ]
         self._test_urls(urls, self._extra_handlers())
 
+    def test_ftp_no_leak(self):
+        # gh-140691: When the data connection (but not control connection)
+        # cannot be made established, we shouldn't leave an open socket object.
+
+        class MockError(OSError):
```

Comment:
Why does this need to be `MockError`? the `except` pieces don't look for MockError specifically